### PR TITLE
Move to vitest

### PR DESCRIPTION
**Pull request by manuelmeister**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,52 +46,42 @@
...
-    "@types/jest": "29.5.12",
     "@typescript-eslint/eslint-plugin": "5.10.1",
     "@typescript-eslint/parser": "5.10.1",
+    "@vitest/coverage-istanbul": "^2.0.5",
+    "@vue/compiler-sfc": "3.3.6",
     "@vue/eslint-config-standard": "6.1.0",
-    "@vue/test-utils": "2.4.1",
-    "axios": "0.25.0",
+    "@vue/test-utils": "2.4.6",
+    "axios": "1.7.4",
```
Comment (BacLuc) [inline package.json:69]: a sneaky update of axios?

File: tests/apiOperations.spec.js [hunk 2]
```diff
@@ -675,6 +676,180 @@ describe('Using dollar methods', () => {
     })
   })
 
+  it('provides find array function on loading embedded collections', async () => {
```
Comment (BacLuc) [inline tests/apiOperations.spec.js:679]: why is this in this PR?

File: tests/store.spec.js [hunk 3]
```diff
@@ -845,6 +843,28 @@ describe('API store', () => {
         )
       })
 
+      it('purges whole store', async () => {
```
Comment (BacLuc) [inline tests/store.spec.js:846]: why are there new tests in this "Move to vitest" PR?

File: tests/store.spec.js [hunk 4]
```diff
@@ -1323,6 +1343,26 @@ describe('API store', () => {
         expect(axiosMock.history.get.length).toEqual(3)
       })
 
+      it('does not delete object from the store if api fails', async () => {
```
Comment (BacLuc) [inline tests/store.spec.js:1346]: why are there new tests in this "Move to vitest" PR?

File: vite.config.js [hunk 5]
```diff
@@ -1,23 +1,41 @@
...
+        }
+      }
     }
   },
-});
+  test: {
+    environment: 'jsdom',
+    globals: true,
+    coverage: {
+      provider: 'istanbul',
```
Comment (BacLuc) [inline vite.config.js:37]: you will have to exclude some folders from the coverage:

in ecamp, we did the following: 
exclude: [...configDefaults.coverage.exclude, '**/src/pdf/**'],

File: tests/store.spec.js (hunk 4 above)
Comment (manuelmeister) [reply tests/store.spec.js:1346]: Because it reported the coverage not exactly the same as before with jest.

File: package.json (hunk 1 above)
Comment (manuelmeister) [reply package.json:69]: We don't have this as a peerDependency. But if you want we can leave it at the outdated version.
It is only used for testing and typing purposes.

File: package.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "hal-json-vuex",
-  "version": "3.0.0-alpha.1",
+  "version": "3.0.0-alpha.2",
```
Comment (usu) [inline package.json:3]: Why this? We should only upgrade the version number when we actually publish a new verison, no?

File: package.json [hunk 7]
```diff
@@ -27,6 +27,7 @@
   },
   "type": "module",
   "files": [
+    "/src",
```
Comment (usu) [inline package.json:30]: Why this addition?

Comment (manuelmeister) [reply package.json:30]: Needs to move to #300

**Comment by usu** [comment]
Tested with ecamp print (devel branch) and everything still works 👍

File: .eslintrc [hunk 8]
```diff
@@ -76,8 +76,5 @@
         ]
       }
     ]
-  },
-  "parserOptions": {
-    "parser": "babel-eslint"
```
Comment (BacLuc) [inline .eslintrc:81]: das haben wir im frontend noch behalten

File: package.json [hunk 9]
```diff
@@ -12,13 +12,13 @@
     }
   },
   "scripts": {
-    "build": "cross-env NODE_ENV=production tsc && vite build",
+    "build": "cross-env NODE_ENV=production tsc --noEmit && vite build",
     "build:watch": "cross-env NODE_ENV=production vite build --watch",
     "build-dev": "cross-env NODE_ENV=development tsc && vite build",
     "clean": "cross-env rimraf dist coverage lib",
-    "coverage": "cross-env npm run build && jest --coverage",
+    "coverage": "cross-env npm run build && vitest --coverage.enabled",
```
Comment (BacLuc) [inline package.json:19]: why this ,enabled?
in the frontend, we don't need this, vitest run --coverage is enough

File: tests/apiOperations.spec.js [hunk 10]
```diff
@@ -622,7 +623,7 @@ describe('Using dollar methods', () => {
       }
     }
     const chapter2Response = {
-      id: 1028,
+      id: 1031,
```
Comment (BacLuc) [inline tests/apiOperations.spec.js:626]: why did you change the id?

File: .eslintrc (hunk 8 above)
Comment (manuelmeister) [reply .eslintrc:81]: Why should I keep it? We already have typescript-eslint

**Review by usu — APPROVED** [review]
